providers/edge/src/airflow/providers/edge/cli/api_client.py

```diff
@@ -103,15 +104,16 @@ def worker_register(
 
 def worker_set_state(
     hostname: str, state: EdgeWorkerState, jobs_active: int, queues: list[str] | None, sysinfo: dict
-) -> list[str] | None:
+) -> WorkerSetStateReturn:
     """Update the state of the worker in the central site and thereby implicitly heartbeat."""
-    return _make_generic_request(
+    result = _make_generic_request(
         "PATCH",
         f"worker/{quote(hostname)}",
         WorkerStateBody(state=state, jobs_active=jobs_active, queues=queues, sysinfo=sysinfo).model_dump_json(
             exclude_unset=True
         ),
     )
+    return WorkerSetStateReturn(**result["__data__"])
```

Comment:
It seems the v2 wrapper "wrongly" serializes the class, therefore this compatibility breaks the EdgeWorker in Airflow 3.
```suggestion
    return WorkerSetStateReturn(**result)
```

Reply:
...and to send correct JSON as response in v2 you need to change providers/edge/src/airflow/providers/edge/worker_api/routes/_v2_routes.py:156 to dump the pydantic class as JSON:
```
        return set_state(worker_name, request_obj, session).model_dump()
```